src/librustc_codegen_ironox/ir/instruction.rs

```diff
@@ -0,0 +1,39 @@
+use context::CodegenCx;
+use ir::type_::Type;
+use ir::value::Value;
+use super::super::ModuleIronOx;
+
+use rustc::ty::layout::Align;
+use rustc_codegen_ssa::traits::BaseTypeMethods;
+
+/// An IronOx instruction.
+#[derive(PartialEq, Debug, Eq, Hash)]
```

Comment:
Is this to avoid passing asm strings around then?

Reply:
Yep. I don't want to generate asm on the fly anyway. I'd like to eventually process the IR in two separate passes (the first pass would calculate how much stack space to allocate for each function etc).

Reply:
Seems sensible